functions/src/routers/groupRouter.ts

```diff
@@ -49,36 +87,126 @@ groupRouter.post("/add", async (req, res) => {
     });
   }
 
-  const groupID=englishName.trim().toLowerCase()
-  // TODO: memberCount 및 members 정의 후에 해당 요청이 문제가 없을경우 
+  const groupID = englishName.trim().toLowerCase();
+  // TODO: memberCount 및 members 정의 후에 해당 요청이 문제가 없을경우
   // 해당 유저의 정보에 groups 항목에 해당 그룹을 추가한다.
   admin
     .firestore()
-    .collection("group")
+    .collection("groups")
     .doc(groupID)
     .set({
       name: name,
       type: type,
       description: description,
       owners: [uid],
       members: [uid],
-      memberCount: 1 // memberCount 그룹 추가 시 1명이므로
-    }).then(async ()=>{
+      memberCount: 1, // memberCount 그룹 추가 시 1명이므로
+    })
+    .then(async () => {
       //문제가 없음
       try {
-        await admin.firestore().collection("users").doc(uid).update({
-          groups: admin.firestore.FieldValue.arrayUnion(groupID)
-        })
-        return res.send("Success")
-      }catch(e) {
-        admin.firestore().collection("group").doc(groupID).delete()
-        return res.status(500).json(e)
+        await admin
+          .firestore()
+          .collection("users")
+          .doc(uid)
+          .update({
+            groups: admin.firestore.FieldValue.arrayUnion(groupID),
+          });
+        return res.send("Success");
+      } catch (e) {
+        admin.firestore().collection("group").doc(groupID).delete();
+        return res.status(500).json(e);
       }
-    })
-  
-    // if(!memberCount){
-    // admin.firestore().collection("users").doc(uid).set({...createdDocument, group: name});
-    // }
+    });
+
+  // if(!memberCount){
+  // admin.firestore().collection("users").doc(uid).set({...createdDocument, group: name});
+  // }
   return;
 });
+groupRouter.get("/:userId/groups", async (req, res) => {
+  const uid = req.decodedToken.uid;
+
+  try {
+    // 데이터베이스에서 해당 유저의 정보 가져오기
+    const userData = (
+      await admin.firestore().collection("users").doc(uid).get()
+    ).data();
+
+    // console.log(uid, userData);
+
+    // DB 에서 해당 데이터를 찾을 수 없는경우.
+    if (!userData)
+      return res.status(404).json({ status: 404, error: "USERDATA NOT FOUND" });
+
+    // 해당 유저 그룹에 그룹 정보가 있는지 확인
+    if (!userData.group)
+      return res
+        .status(403)
+        .json({ status: 403, error: "USERGROUP FORBIDDEN", uid });
```

Comment:
권한이 없는 것이 아닌 존재하지 않는 것이기 때문에 빈 그룹 정보인 `[]` 같은 것을 반환해야할 것 같습니다. 
403 을 반환할경우 Frontend 에 추가적인 불필요한 Error handling 을 해야합니다.

```suggestion
        .json({ status: 200, [] });
```

Reply:
Error handling이라는 것이 Frontend에서 try catch(e)할때의 catch(e)인가요?

Reply:
그리고 
```
if (!userData.groups){
      return res
        .status(200)
        .json({ status: 200, [] })}
```
vscode로 보았을 때 해당 부분에서 ]}여기에 오류 뜹니다

Reply:
<img width="402" alt="addnotify error" src="https://user-images.githubusercontent.com/74362214/106588084-b9fb9980-658d-11eb-9e3a-9fca435489a3.PNG">

Expression expected .랑 저겁니다.